integration-tests/test/dag.py

```diff
@@ -0,0 +1,23 @@
+from collections import defaultdict
+from typing import Dict, Set
+
+class Dag():
```

Comment:
I'd just use plain dict with type annotations instead of introducing a class.  The class doesn't bring anything valuable to the table.

Reply:
ok